### ISSUE TITLE
a problem about loss

### ISSUE THREAD

**Issue by zxxv11** [J-Cuadrado/OF_EV_SNN#2]
### loss
Could you please tell me why loss reach to nan in every epoch?
![image](https://user-images.githubusercontent.com/47053037/236419919-8717f61c-e8f3-4043-b91b-2bea29ed878a.png)

**Comment by J-Cuadrado** [comment]
Hello, that problem seems to appear when not enough spikes are triggered during the forward pass. In that case, the weights seems to explode to a nan value, and therefore subsequent predictions automatically show nan values everywhere. So far, we have found that playing with the MultiplyFactor value is an effective way to counteract this problem, since it encourages activity within the network. I hope this is of help to you, do not hesitate to contact us for further information.

**Comment by zig-zagreus** [comment]
Hi, sorry to ask another question but how should i be formatting my pathway for the line:
valid_dataset = DSECDatasetLite(root =/opt/Partage/DSEC/saved_flow_data
so that i can call the valid split doubleseq, for the test_network_metrics file
Ive tried both the full user pathway and only the pathway after the OF_EV_SNN-main but Im getting the no file found error and also the Syntax Error 
:(unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape

any advice would be super appreaciated as its my first time working with neural networks! :)

**Comment by J-Cuadrado** [comment]
Hello!

It is hard to know what is going wrong with your code without actually
knowing your data saving structure, but here are some tips:
1) It is easier to feed the algorithm the global path of your data (if you
copy the github repository as is, it should be something like
/home/user/OF_EV_SNN/data/dataset/saved_flow_data. This is true both for
the train and for the test scripts.
2) For the saved data, everything should be in the right place if you have
executed the script dsec_dataset_lite/main.py as it is.

Do not hesitate to let me know if you experience any more issues, I will be
glad to help you as much as I can.

Kind regards,

Javier CUADRADO

**Comment by zxxv11** [comment]
I have run dsec_dataset_lite/main.py and generated data files in the corresponding directory. When training, the loss will be nan. I don’t know if it is the problem of the DSEC dataset I downloaded or other problems.

**Comment by J-Cuadrado** [comment]
There should not be any problem concerning the DSEC dataset... Have you tried increasing the MultiplyFactor on line 80?

`net = NeuronPool_Separable_Pool3d(multiply_factor = 25.).to(device)`

With a multiplicative factor of around 50, you should not be encountering that problem anymore, please let us know if the problem persists.

**Comment by zxxv11** [comment]
> 关于 DSEC 数据集应该没有任何问题......您是否尝试增加第 80 行的 MultiplyFactor？
> 
> `net = NeuronPool_Separable_Pool3d(multiply_factor = 25.).to(device)`
> 
> 乘数约为 50，您应该不会再遇到该问题，如果问题仍然存在，请告诉我们。

Thank you，I try making multiply_factor = 25.，my problem probably be solved.

**Comment by zxxv11** [comment]
> 关于DSEC 数据集应该没有任何问题......您是否尝试增加第80行的MultiplyFactor？
> 
> `net = NeuronPool_Separable_Pool3d(multiply_factor = 25.).to(device)`
> 
> 乘数大约为50，您应该不会再遇到该问题，如果问题仍然存在，请告诉我们。

Thank you，I try making multiply_factor = 50.，my problem probably be solved.

**Comment by J-Cuadrado** [comment]
I will mark this issue as solved then, do not hesitate to contact us again for further questions
